Utils/FaviconFetcher.swift

```diff
@@ -93,73 +94,84 @@ public class FaviconFetcher : NSObject, NSXMLParserDelegate {
+                if let urlParam = refreshNode?.attributes["content"],
+                    let index = urlParam.rangeOfString("URL="),
+                    let url = NSURL(string: urlParam.substringFromIndex(index.startIndex.advancedBy(4))) {
+                        reloadUrl = url
                     }
+            } catch {}
 
-                    guard let href = link.attribute("href") where iconType != nil else {
-                        return
-                    }
 
-                    if (href.endsWith(".ico")) {
-                        iconType = .Guess
-                    }
+            if let url = reloadUrl {
+              return self.parseHTMLForFavicons(url)
+            }
 
-                    if let type = iconType where !bestType.isPreferredTo(type),
-                        let iconUrl = NSURL(string: href, relativeToURL: url) {
-                            let icon = Favicon(url: iconUrl.absoluteString, date: NSDate(), type: type)
-                            // If we already have a list of Favicons going already, then add it…
-                            if (type == bestType) {
-                                icons.append(icon)
-                            } else {
-                                // otherwise, this is the first in a new best yet type.
-                                icons = [icon]
-                                bestType = type
-                            }
+            //map all the icon types to an array of Favicons
+            let headElement = element["html"]["head"]["link"]
+            let relAttributes = ["shortcut icon","icon","apple-touch-icon","apple-touch-icon-precomposed"]
+            let iconsMapped = relAttributes.flatMap({ (relIcon:String) -> Favicon? in
+                do {
+                    guard let iconHref = try headElement.withAttr("rel", relIcon).element?.attributes["href"] else {
+                        return nil
+                    }
+                    if let iconUrl = NSURL(string: iconHref, relativeToURL: url) {
+                        let type = iconHref.endsWith(".ico") ? IconType.Guess : self.iconTypeForLink(relIcon)
+                        let icon = Favicon(url: iconUrl.absoluteString, date: NSDate(), type: type)
+                        return icon
                     }
+                } catch {
+                    return nil
                 }
+                return nil
+            })
 
-                // If we haven't got any options icons, then use the default at the root of the domain.
-                if let url = NSURL(string: "/favicon.ico", relativeToURL: url) where icons.isEmpty {
-                    let icon = Favicon(url: url.absoluteString, date: NSDate(), type: .Guess)
-                    icons = [icon]
-                }
+            //sort that array such that the first Item is the perferred type
+            icons = iconsMapped.sort({ (a:Favicon, b:Favicon) -> Bool in
```

Comment:
Here I sort the Favicons based on the types. This is equivalent to the "bestType" the old code kept track of.